sql/src/main/java/io/crate/operation/reference/sys/check/cluster/TablesNeedUpgradeSysCheck.java

```diff
@@ -0,0 +1,160 @@
+    private static final String STMT = "select schema_name || '.' || table_name, min_lucene_version " +
+                                       "from sys.shards where min_lucene_version <> '" +
+                                       org.apache.lucene.util.Version.LATEST.toString() + "'";
+    private static final String STMT_NOT_IN = " AND schema_name || '.' || table_name NOT IN (";
+    private static final String STMT_SUFFIX = " order by 1";
+    private static final int LIMIT = 50_000;
+
+    private final SQLOperations.Session session;
+    private final ClusterService clusterService;
+    private final Schemas schemas;
+    private volatile Collection<String> tablesNeedUpgrade;
+
+    @Inject
+    public TablesNeedUpgradeSysCheck(ClusterService clusterService, Schemas schemas, SQLOperations sqlOperations) {
+        super(ID, DESCRIPTION, Severity.MEDIUM);
+        this.clusterService = clusterService;
+        this.schemas = schemas;
+        this.session = sqlOperations.createSession("sys", Option.NONE, LIMIT);
+        this.session.parse(SQLOperations.Session.UNNAMED, STMT, Collections.emptyList());
+            sqlOperations.createSession(null, Option.NONE, 1);
+    }
+
+    @Override
+    public BytesRef description() {
+        String linkedDescriptionBuilder = DESCRIPTION + tablesNeedUpgrade + ' ' + LINK_PATTERN + ID;
+        return new BytesRef(linkedDescriptionBuilder);
+    }
+
+    @Override
+    public boolean validate() {
+        List<String> tablesNeedReindexing = LuceneVersionChecks.tablesNeedReindexing(
+            schemas,
+            clusterService.state().metaData());
+
+        String statement;
+        if (tablesNeedReindexing.isEmpty()) {
+            statement = STMT + STMT_SUFFIX;
+        } else {
+            StringBuilder sb = new StringBuilder(STMT + STMT_NOT_IN);
+            for (int i = 0; i < tablesNeedReindexing.size(); i++) {
+                sb.append("'").append(tablesNeedReindexing.get(i)).append("'");
+                if (i < tablesNeedReindexing.size() - 1) {
+                    sb.append(",");
+                }
+            }
+            statement = sb.toString();
+        }
+
+        final CompletableFuture<Collection<String>> result = new CompletableFuture<>();
+        try {
+            session.parse(SQLOperations.Session.UNNAMED, statement, Collections.emptyList());
+            session.bind(
+                SQLOperations.Session.UNNAMED,
+                SQLOperations.Session.UNNAMED,
+                Collections.singletonList(tablesNeedReindexing),
+                null);
+            session.execute(
+                SQLOperations.Session.UNNAMED,
+                0,
+                new SycCheckResultReceiver(result));
+            session.sync();
+        } catch (Throwable t) {
+            result.completeExceptionally(t);
+        }
+        try {
+            tablesNeedUpgrade = result.get(5, TimeUnit.SECONDS);
```

Comment:
just to give you an idea what kind of impact this can have:

```

↪ cr8 timeit -s "select * from sys.checks" --hosts localhost:4201 -c 50 
Runtime (in ms):
    mean:    4.911 ± 1.778
    min/max: 0.855 → 20.724
Percentile:
    50:   3.219 ± 4.969 (stdev)
    95:   15.350
    99.9: 20.724


↪ cr8 timeit -s "select * from sys.checks" --hosts localhost:4200 -c 50 
Runtime (in ms):
    mean:    1012.253 ± 727.904
    min/max: 2.737 → 5034.469
Percentile:
    50:   10.640 ± 2034.131 (stdev)
    95:   5013.495
    99.9: 5034.469
```